tt_metal/llrt/tt_cluster.cpp

```diff
@@ -434,15 +434,15 @@ void Cluster::read_dram_vec(
 }
 
 void Cluster::write_core(
-    const void *mem_ptr, uint32_t sz_in_bytes, tt_cxy_pair core, uint64_t addr, bool small_access) const {
+    const void *mem_ptr, uint32_t sz_in_bytes, tt_cxy_pair core, uint64_t addr, bool small_access, std::string tlb_to_use) const {
```

Comment:
nit but maybe rename `tlb_to_use` to `dynamic_fallback_tlb`

Reply:
it is crazy that we are passing a string to determine a tlb (I know that's not new), but this string should at least be a const reference so we don't malloc every time we write a tlb.

Reply:
Actually I just noticed we have `Cluster::write_reg` that uses `REG_TLB` that we can use instead of adding the string arg to `Cluster::write_core`. Going to switch to that instead.

Reply:
I've switched to write_reg, so there are no changes to write_core anymore